src/jit/lsra.cpp

```diff
@@ -8536,6 +8545,13 @@ void LinearScan::recordMaxSpill()
         maxSpill[TYP_FLOAT] += 1;
     }
 #endif // _TARGET_X86_
+
+    if (compiler->opts.compUseSoftFP)
+    {
+        JITDUMP("Adding a spill temp for moving target address to a register.\n");
+        maxSpill[TYP_INT] += 1;
```

Comment:
> Eventually the call address goes to r8 though. And this probably requires one more temp variable.

There should be a spill temp accounted for with the spill address of the target register. Given the dump segment that you shared, it would seem that:
- The instruction at N1861 defines a value (I981) into r0. It is spilled, so that will result in `maxSpill[TYP_INT]` being set to at least 1 (it would be more if there are other outstanding spills, but that's not 100% clear from the dump fragment).
- The instruction at N1863 defines the call target address (and the definition is located at 1864). It is also spilled, so at this point `maxSpill[TYP_INT]` should be at least 2.
- It appears that the call is at N1865. It reloads I981 into r0, which causes `maxSpill[TYP_INT]` to be decremented.
- It then reloads the target address (C983) into r8, which causes `maxSpill[TYP_INT]` to be decremented again - back to 0 or whatever it was at Location 1861.
- Then, it spills r12 (V7), which requires an additional spill temp, so `maxSpill[TYP_INT]` should now be at least one again.

During codegen, these spills and reloads should be happening in this same order. Can you verify that's the case?